homeassistant/components/igloohome/sensor.py

```diff
@@ -0,0 +1,72 @@
+"""Implementation of the sensor platform."""
+
+from datetime import timedelta
+import logging
+
+from aiohttp import ClientError
+from igloohome_api import Api as IgloohomeApi, ApiException, GetDeviceInfoResponse
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import IgloohomeConfigEntry
+from .entity import IgloohomeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SCAN_INTERVAL = timedelta(hours=1)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: IgloohomeConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensor entities."""
+
+    api = entry.runtime_data[0]
+    devices = entry.runtime_data[1]
+
+    async_add_entities(
+        (
+            IgloohomeBatteryEntity(
+                api_device_info=device,
+                api=api,
+            )
+            for device in devices
+            if device.batteryLevel is not None
+        ),
+        update_before_add=True,
+    )
+
+
+class IgloohomeBatteryEntity(IgloohomeBaseEntity, SensorEntity):
```

Comment:
are you really going to add only Battery as sensor (in the foreseeable future)?

Reply:
@zweckj , yes. Batteries are the only sensor entity available from the company's API and was the simpliest integration available that is a useful feature. The company's main products are battery-powered locks and so I plan to add the Lock entity down the road. However, that feature is more complicated and did not seem like the best option for a first integration.